src/server/db_slice.cc

```diff
@@ -1449,13 +1427,29 @@ pair<uint64_t, size_t> DbSlice::FreeMemWithEvictionStep(DbIndex db_ind, size_t s
 
           // returns when whichever condition is met first
           if ((evicted_items == max_eviction_per_hb) || (evicted_bytes >= increase_goal_bytes))
-            return return_cb();
+            goto finish;
         }
       }
     }
+  }  // FiberAtomicGuard
+
+finish:
```

Comment:
I replaced this `goto` in the past :face_with_diagonal_mouth: I wonder if we could somehow still get rid of it but with this double for-loop seems like inevitable

Reply:
There is nothing wrong in goto statements, it's not a taboo. But we should not use them if there are clearer tools.